frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx

```diff
@@ -0,0 +1,428 @@
+        receivers.splice(index, 1, newReceiver);
+      } else {
+        receivers.push(newReceiver);
+      }
+      return receivers;
+    });
+
+    // Update Route & RouteLabels
+    if (isDefaultReceiver) {
+      _.set(route, 'receiver', newReceiver.name);
+    }
+
+    const newRoute = _.isEmpty(formValues.routeLabels)
+      ? undefined
+      : createRoute(newReceiver, formValues.routeLabels);
+    _.update(route, 'routes', (routes = []) => {
+      if (editReceiverNamed) {
+        const index = _.findIndex(routes, { receiver: editReceiverNamed });
+        if (index !== -1) {
+          if (!newRoute) {
+            // no routing labels for receiver, remove old route
+            routes.splice(index, 1);
+          } else {
+            // update receiver's route with new route/labels
+            routes.splice(index, 1, newRoute);
+          }
+        } else if (newRoute) {
+          // receiver didn't have a prior route, so add new route
+          routes.push(newRoute);
+        }
+      } else if (newRoute) {
+        // add route for new receiver
+        routes.push(newRoute);
+      }
+      return routes;
+    });
+
+    // Update 'alertmanager-main' Secret with new alertmanager.yaml configuration
+    setInProgress(true);
+    patchAlertmanagerConfig(secret, config).then(
+      () => {
+        setErrorMsg('');
+        setInProgress(false);
+        history.push('/monitoring/alertmanagerconfig');
+      },
+      (err) => {
+        setErrorMsg(err.message);
+        setInProgress(false);
+      },
+    );
+  };
+
+  return (
+    <div className="co-m-pane__body co-m-pane__form">
+      <Helmet>
+        <title>{titleVerb} Receiver</title>
+      </Helmet>
+      <form className="co-m-pane__body-group" onSubmit={save}>
+        <h1 className="co-m-pane__heading">
+          {titleVerb} {isDefaultReceiver && 'Default'} Receiver
+        </h1>
+        <div className="form-group">
+          <label className="control-label co-required">Receiver Name</label>
+          <input
+            className="pf-c-form-control"
+            type="text"
+            value={formValues.receiverName}
+            onChange={(e) =>
+              dispatchFormChange({
+                type: 'handleChangeEvent',
+                payload: { ...e },
+              })
+            }
+            aria-describedby="receiver-name-help"
+            name="receiverName"
+            data-test-id="receiver-name"
+            required
+          />
+        </div>
+        <div className="form-group co-m-pane__dropdown">
+          <label className="control-label co-required">Receiver Type</label>
+          <Dropdown
+            title={'Select Receiver Type...'}
+            name="receiverType"
+            items={receiverTypes}
+            dropDownClassName="dropdown--full-width"
+            data-test-id="receiver-type"
+            selectedKey={formValues.receiverType}
+            onChange={(e) =>
```

Comment:
This isn't an event, so `e` as a name is misleading.

```suggestion
            onChange={(receiverType) =>
```

Reply:
fixed